src/box/errcode.h

```diff
@@ -354,34 +403,33 @@ struct errcode_record {
  */
 
 ENUM(box_error_code, ERROR_CODES);
-extern struct errcode_record box_error_codes[];
+extern const struct errcode_record box_error_codes[];
+
+/** Record for unknown errocode. */
+extern const struct errcode_record errcode_record_unknown;
 
-static inline bool
-tnt_errcode_is_defined(uint32_t errcode)
+/** Return the error fields info for the errorcode. */
+static inline const struct errcode_record *
+tnt_errcode_record(uint32_t errcode)
 {
-	return errcode < box_error_code_MAX &&
-	       box_error_codes[errcode].errstr != NULL;
+	if (errcode >= box_error_code_MAX ||
+	    box_error_codes[errcode].errstr == NULL)
+		return &errcode_record_unknown;
+	return &box_error_codes[errcode];
```

Comment:
I think this belongs in the first commit. Introduce `tnt_errcode_is_defined` just to drop it looks pointless.

Reply:
Done.